src/components/Equity/Equity.tsx

```diff
@@ -0,0 +1,159 @@
+  selectPortfolioLoading,
+  selectUnderlyingLpAssetsWithBalancesAndIcons,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { EquityAccountRow } from './EquityAccountRow'
+import { EquityLpRow } from './EquityLpRow'
+import { EquityRowLoading } from './EquityRow'
+import { EquityStakingRow } from './EquityStakingRow'
+import { UnderlyingAsset } from './UnderlyingAsset'
+
+type EquityProps = {
+  assetId: AssetId
+  accountId?: AccountId
+}
+
+export const Equity = ({ assetId, accountId }: EquityProps) => {
+  const translate = useTranslate()
+  const portfolioLoading = useSelector(selectPortfolioLoading)
+  const opportunitiesLoading = useAppSelector(selectOpportunityApiPending)
+  const isLoading = portfolioLoading || opportunitiesLoading
+  const assets = useAppSelector(selectAssets)
+  const asset = assets[assetId]
+  const borderColor = useColorModeValue('blackAlpha.50', 'whiteAlpha.50')
+  const filter = useMemo(() => {
+    return {
+      assetId,
+      ...(accountId ? { accountId } : {}),
+    }
+  }, [accountId, assetId])
+
+  const totalEquityBalance = useAppSelector(state => selectEquityTotalBalance(state, filter))
+  const equityRows = useAppSelector(state => selectEquitiesFromFilter(state, filter))
+
+  const { amountCryptoPrecision: totalCryptoHumanBalance, fiatAmount: totalFiatBalance } =
+    totalEquityBalance
+
+  const lpAssetBalanceFilter = useMemo(
+    () => ({
+      assetId,
+      accountId,
+      lpId: assetId as LpId,
+    }),
+    [accountId, assetId],
+  )
+
+  const underlyingAssetsWithBalancesAndIcons = useAppSelector(state =>
+    selectUnderlyingLpAssetsWithBalancesAndIcons(state, lpAssetBalanceFilter),
+  )
+
+  const renderEquityRows = useMemo(() => {
+    if (portfolioLoading)
+      return Array.from({ length: 4 }).map((_, index) => (
+        <EquityRowLoading key={`eq-row-loading-${index}`} />
+      ))
+    return equityRows.map(item => {
+      switch (item.type) {
+        case AssetEquityType.Staking:
+          return (
+            <EquityStakingRow
+              key={item.id}
+              assetId={assetId}
+              opportunityId={item.id as OpportunityId}
+              totalFiatBalance={totalFiatBalance}
+              color={item.color}
+              accountId={accountId}
+            />
+          )
+        case AssetEquityType.LP:
+          return (
+            <EquityLpRow
+              key={item.id}
+              assetId={assetId}
+              opportunityId={item.id as OpportunityId}
+              totalFiatBalance={totalFiatBalance}
+              color={item.color}
+              accountId={accountId}
+            />
+          )
+        case AssetEquityType.Account:
+          return (
+            <EquityAccountRow
+              key={item.id}
+              assetId={assetId}
+              accountId={item.id as AccountId}
+              totalFiatBalance={totalFiatBalance}
+              color={item.color}
```

Comment:
This is fine since we only have 3 variants - usually would be inclined towards not repeating passing the exact same props down (except for `opportunityId` for staking and lp which isn't present in account equity type)